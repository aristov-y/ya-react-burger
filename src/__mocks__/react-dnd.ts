const mockDropRef = jest.fn();
const mockDragRef = jest.fn();

const useDrag = jest.fn().mockImplementation(() => [undefined, mockDragRef]);
const useDrop = jest.fn().mockImplementation(() => [undefined, mockDropRef]);

export {
  useDrag,
  useDrop
}
