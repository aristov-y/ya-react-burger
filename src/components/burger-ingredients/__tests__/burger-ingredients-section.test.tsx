import React from 'react';
import { create, act, ReactTestRenderer } from 'react-test-renderer';
import BurgerIngredientsSection from '../burger-ingredients-section';
import { useSelector } from 'react-redux';
import { useDrag } from 'react-dnd';

jest.mock('../../burger-ingredient')

const onShowDetails = jest.fn();

const items: any[] = [{
  _id: 'Bun01',
}, {
  _id: 'Main01'
}, {
  _id: 'Main02'
}, {
  _id: 'Sauce01'
}];

const store = {
  constructor: {
    bun: {
      _id: 'Bun01'
    },
    main: [{
      _id: 'Main01'
    }, {
      _id: 'Sauce01'
    }, {
      _id: 'Sauce01'
    }]
  }
}

beforeEach(() => {
  (useSelector as jest.Mock).mockImplementation((fn) => fn(store));
  (useDrag as jest.Mock).mockImplementation(() => []);
});

describe('burger-ingredients-section', function () {
  it('should render', function () {
    let component: ReactTestRenderer;
    act(() => {
      component = create(<BurgerIngredientsSection
        onShowDetails={onShowDetails}
        title={'Title'}
        items={items}
      />);
    });

    expect(component.toJSON()).toMatchSnapshot();
  });
});
