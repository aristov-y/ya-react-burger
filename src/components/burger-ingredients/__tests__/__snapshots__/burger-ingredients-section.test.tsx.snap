// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`burger-ingredients-section should render 1`] = `
Array [
  <div>
    <span
      className="text text_type_main-medium"
    >
      Title
    </span>
  </div>,
  <div
    className="mt-10 pt-4 pl-4"
  >
    BurgerIngredients
    BurgerIngredients
    BurgerIngredients
    BurgerIngredients
  </div>,
]
`;
